src/fetch.js

```diff
@@ -5,6 +5,8 @@ const Request = require('./request')
 const Response = require('./response')
 const { STATUS_CODES } = require('http')
 
+const kAgent = Symbol('agent')
```

Comment:
I would propose a symbols file which is not accessible for end users.

Reply:
okay and then add back the `.close` method to fetch in case a user wants to use that?

Reply:
IMO expose a proper public api or make it entirely hidden.